linkerd/main/src/main/scala/io/buoyant/Linkerd.scala

```diff
@@ -44,6 +44,16 @@ object Linkerd extends App {
           running.servers.map { server =>
             log.info("serving %s on %s:%d", server.router, server.ip, server.port)
             val listening = server.serve()
+            for (name <- server.announce) {
+              val announcers = running.announcers.filter {
+                case (prefix, announcer) => name.startsWith(prefix)
+              }
+              for ((prefix, announcer) <- announcers) {
+                log.info("announcing %s as %s to %s", server.addr, name.show, announcer.scheme)
+                announcer.announce(server.addr, name.drop(prefix.size)).onSuccess(closeOnExit)
+              }
+              if (announcers.isEmpty) log.warning("no announcer found for %s", name.show)
+            }
```

Comment:
Thinking aloud here -- perhaps it makes sense to pass announcers to routers/servers rather than have servers expose their paths to main.

My thought here is that it's an implementation detail that the list of endpoints is statically provided.  We could envision a model where routers can register additional services at runtime.  If routers have access to the announcers -- or an "AnnouncerDispatcher" that deals with prefix mapping/removal -- then the static-ness of announcement is an implementation detail rather than an fundamental assumption.

Perhaps YAGNI.  I'm fine with this as-is, just thinking about some possible future use cases.